test/functional/test_framework/wallet.py

```diff
@@ -119,12 +119,21 @@ def _bulk_tx(self, tx, target_weight):
     def get_balance(self):
         return sum(u['value'] for u in self._utxos)
 
-    def rescan_utxos(self):
+    def rescan_utxos(self, *, include_mempool=True):
         """Drop all utxos and rescan the utxo set"""
         self._utxos = []
         res = self._test_node.scantxoutset(action="start", scanobjects=[self.get_descriptor()])
         assert_equal(True, res['success'])
+        spents = set()
+        for txid in self._test_node.getrawmempool():
+            tx = self._test_node.getrawtransaction(txid=txid, verbose=True)
+            if include_mempool:
+                self.scan_tx(tx)
+            else:
+                spents.update([(i["txid"], i["vout"]) for i in tx["vin"]])
```

Comment:
This will consider the mempool if `!include_mempool`, which may be confusing

Reply:
What's proper way that you might consider reasonable?

Reply:
No, just remove the else branch

Reply:
If I remove the else branch `test_sequence_lock_confirmed_inputs` will encounter `txn-mempool-conflict`.

Reply:
Even if you move the mempool scan last? See https://github.com/bitcoin/bitcoin/pull/26657#discussion_r1049630603

Reply:
You are so right..